src/modules/0.0.22/aws_security_group/index.ts

```diff
@@ -304,9 +300,7 @@ class SecurityGroupMapper extends MapperBase<SecurityGroup> {
         // You can mess with its rules, but not this record itself, so any attempt to update it
         // is instead turned into *restoring* the value in the database to match the cloud value
         // Check if there is a VPC for this security group in the database
-        const vpcDbRecord = (await awsVpcModule.vpc.db.read(ctx)).find(
-          (a: any) => a.vpcId === e.vpc?.vpcId && a.region === e.region,
-        );
+        const vpcDbRecord = await awsVpcModule.vpc.db.read(ctx, super.generateId(e.vpc?.vpcId, e.region));
```

Comment:
Interesting, yes that might be helpful. I was just trying to get rid of the manual work of `${}|${}`

Reply:
The only thing I do not have clear in my mind and I'd need to see how to do it is the type enforce. Also, I might leave a concat helper around because I was using this function also in a custom `entityId` definition.

Reply:
Maybe the ones that manually define `entityId` can also manually define their own `generateId`?